### PR TITLE
dm/load: fix concurrent call Loader.Status

### LINKED ISSUE #3457: load unit will panic if query-status too often
### What did you do?

in CI

https://ci2.pingcap.net/blue/organizations/jenkins/dm_ghpr_integration_test/detail/dm_ghpr_integration_test/626/pipeline

### What did you expect to see?

success 

### What did you see instead?

```
[2021-11-15T07:53:35.989Z] WARNING: DATA RACE
[2021-11-15T07:53:35.989Z] Read at 0x00c00061f6e0 by goroutine 291:
[2021-11-15T07:53:35.989Z]   runtime.mapaccess1_faststr()
[2021-11-15T07:53:35.989Z]       /usr/local/go/src/runtime/map_faststr.go:12 +0x0
[2021-11-15T07:53:35.989Z]   github.com/pingcap/ticdc/dm/loader.(*Loader).printStatus()
[2021-11-15T07:53:35.989Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/loader/status.go:51 +0x304
[2021-11-15T07:53:35.989Z] 
[2021-11-15T07:53:35.989Z] Previous write at 0x00c00061f6e0 by goroutine 368:
[2021-11-15T07:53:35.989Z]   runtime.mapassign_faststr()
[2021-11-15T07:53:35.989Z]       /usr/local/go/src/runtime/map_faststr.go:202 +0x0
[2021-11-15T07:53:35.989Z]   github.com/pingcap/ticdc/dm/loader.(*Loader).printStatus()
[2021-11-15T07:53:35.989Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/loader/status.go:52 +0x3c6
[2021-11-15T07:53:35.989Z] 
[2021-11-15T07:53:35.989Z] Goroutine 291 (running) created at:
[2021-11-15T07:53:35.989Z]   github.com/pingcap/ticdc/dm/loader.(*Loader).Status()
[2021-11-15T07:53:35.989Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/loader/status.go:37 +0x2c9
[2021-11-15T07:53:35.989Z]   github.com/pingcap/ticdc/dm/dm/worker.(*SubTask).unitTransWaitCondition()
[2021-11-15T07:53:35.989Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/dm/worker/subtask.go:651 +0x57e
[2021-11-15T07:53:35.989Z]   github.com/pingcap/ticdc/dm/dm/worker.(*SubTask).run()
[2021-11-15T07:53:35.989Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/dm/worker/subtask.go:240 +0x124
[2021-11-15T07:53:35.989Z]   github.com/pingcap/ticdc/dm/dm/worker.(*SubTask).fetchResultAndUpdateStage()
[2021-11-15T07:53:35.989Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/dm/worker/subtask.go:329 +0x1184
[2021-11-15T07:53:35.989Z] 
[2021-11-15T07:53:35.989Z] Goroutine 368 (running) created at:
[2021-11-15T07:53:35.989Z]   github.com/pingcap/ticdc/dm/loader.(*Loader).Status()
[2021-11-15T07:53:35.989Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/loader/status.go:37 +0x2c9
[2021-11-15T07:53:35.989Z]   github.com/pingcap/ticdc/dm/dm/worker.(*SubTask).Status()
[2021-11-15T07:53:35.989Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/dm/worker/status.go:31 +0xa3
[2021-11-15T07:53:35.989Z]   github.com/pingcap/ticdc/dm/dm/worker.(*SubTask).StatusJSON()
[2021-11-15T07:53:35.989Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/dm/worker/status.go:38 +0x6d
[2021-11-15T07:53:35.989Z]   github.com/pingcap/ticdc/dm/dm/worker.(*SubTask).fetchResultAndUpdateStage()
[2021-11-15T07:53:35.989Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/dm/worker/subtask.go:315 +0x6d7
[2021-11-15T07:53:35.989Z] ==================
```

and finally

```
[2021-11-15T07:53:35.989Z] fatal error: concurrent map writes
```

### Versions of the cluster

DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):
```console
5.3.0
```
Upstream MySQL/MariaDB server version:
```console
(paste upstream MySQL/MariaDB server version here)
```
Downstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):
```console
(paste TiDB cluster version here)
```
How did you deploy DM: tiup or manually?
```console
(leave TiUP or manually here)
```
Other interesting information (system version, hardware config, etc):
```console
>
>
```

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
close #3457
close https://github.com/pingcap/ticdc/issues/3252

### What is changed and how it works?
Loader.Status is invoked by RPC, so we should not assume it's not concurrent

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
Fix a bug when too often query status is called in Load unit, DM-worker may encounter data race or panic
```

### CODE REVIEW

**Comment by GMHDBJD** [comment]
Maybe this pr also fix this issue https://github.com/pingcap/ticdc/issues/3252

File: dm/loader/loader.go [hunk 1]
```diff
@@ -441,8 +441,8 @@ type Loader struct {
 	// to calculate remainingTimeGauge metric, map will be init in `l.prepare.prepareDataFiles`
 	dbTableDataTotalSize        map[string]map[string]*atomic.Int64
 	dbTableDataFinishedSize     map[string]map[string]*atomic.Int64
-	dbTableDataLastFinishedSize map[string]map[string]int64
-	dbTableDataLastUpdatedTime  time.Time
+	dbTableDataLastFinishedSize map[string]map[string]*atomic.Int64
```
Comment (D3Hunter) [inline dm/loader/loader.go:444]: would better to implement in object oriented way
```go
map[string]map[string]*tableData
type tableData struct {
  total atomic.Int64
  finished atomic.Int64
  lastFinished atomic.Int64
}
```

Comment (lance6716) [reply dm/loader/loader.go:444]: we're going to deprecate Loader soon, currently I only want to fix bugs with least effort 😂

**Review by D3Hunter — APPROVED** [review]

**Review by Ehco1996 — APPROVED** [review]
LGTM

**Pull request merged**